syntax = "proto3";

message LocationMessage {
  int64 id = 1;
  int64 person_id = 2;
  string longitude = 3;
  string latitude = 4;
  string creation_time = 5;
}

message Empty {
}

message LocationIdMessage {
  string id = 1;
}
message LocationMessageList {
  repeated LocationMessage locations = 1;
}

message FindByPersonAndDatePrm {
  int64 person_id = 1;
  string start_date = 2;
  string end_date = 3;
}

message FindByDistancePrm {
  int64 person_id = 1;
  string longitude = 2;
  string latitude = 3;
  int32 meters = 4;
  string start_date = 5;
  string end_date = 6;
}

service LocationService {
    rpc Create(LocationMessage) returns (LocationMessage);
    rpc Get(Empty) returns (LocationMessageList);
    rpc GetDetail(LocationIdMessage) returns (LocationMessage);
    rpc FindByPersonAndDate(FindByPersonAndDatePrm) returns (LocationMessageList);
    rpc FindByDistance(FindByDistancePrm) returns (LocationMessageList);
}